src/app/exercise/exercise.page/exercise.page.spec.ts

```diff
@@ -0,0 +1,410 @@
+import { ExercisePage } from './exercise.page';
+import { RouterTestingModule } from '@angular/router/testing';
+import { ExerciseSettingsDataMockService } from '../../services/exercise-settings-data.mock.service';
+import { ExerciseMockService } from '../exercise.mock.service';
+import { ModalModule } from '../../shared/modal/modal.module';
+import { MockExercise } from '../MockExercise';
+import { SharedComponentsModule } from '../../shared/components/shared-components/shared-components.module';
+import {
+  TestBed,
+  fakeAsync,
+  flush,
+} from '@angular/core/testing';
+import { ActivatedRoute } from '@angular/router';
+import { ExerciseService } from '../exercise.service';
+import { timeoutAsPromise } from '../../shared/ts-utility';
+import { PlayerMockService } from '../../services/player.mock.service';
+import { ExerciseModule } from '../exercise.module';
+import {
+  PlayerService,
+  NoteEvent,
+} from '../../services/player.service';
+import { ExercisePageDebugger } from './exerice.page.debugger.spec';
+import { TestingUtility } from '../../shared/testing-utility';
+
+describe('ExercisePage', () => {
+  const spies: jasmine.Spy[] = [];
+  let exercisePageDebugger: ExercisePageDebugger;
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      imports: [
+        ExerciseModule,
+        ModalModule,
+        // IonicModule.forRoot({
```

Comment:
Commented out because of this issue: https://github.com/ShacharHarshuv/open-ear/issues/75